packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -122,8 +122,9 @@ export async function fetchHelper(
         };
     }, (error) => {
         const online = isOnline();
-        const errorText = `${error}`;
-
+        let errorText = `${error}`;
+        const spoRegex = /(http.*\.sharepoint(-df)*\.com)/;
+        errorText = errorText.toLowerCase().replace(spoRegex, "REDACTED (sharepoint url)");
```

Comment:
Should we use non-greedy matches (`.*?`) here? Not sure how much of a perf improvement would we get for non-degenerate inputsbut seems like a good practice unless we explicitly need greedy behavior. And the second one even seems like it could be `(-df)?`, we only expect it to appear 0 or 1 times, right?

And I think we can do the regex match case insensitive without having to call `toLowerCase()` on the text itself, `const spoRegex = /.../i;`, that would keep the rest of the error just like it looks right now.

Reply:
I'd suggest detecting any http:// or https:// link and get rid of it (i.e. not limit it to SharePoint only)